ui/store/actions.js

```diff
@@ -51,9 +52,37 @@ import * as actionConstants from './actionConstants';
 
 let background = null;
 let promisifiedBackground = null;
+const retryQueue = [];
+
+const submitRequestToBackground = async (method, args) => {
+  if (isManifestV3) {
+    if (background.connectionStream.readable) {
+      await promisifiedBackground[method](...args);
+    } else {
+      retryQueue.push({ method, args });
+    }
+  } else {
+    await promisifiedBackground[method](...args);
+  }
+};
+
+const clearRetryQueue = () => {
+  while (retryQueue.length) {
+    const item = retryQueue.shift();
+    submitRequestToBackground(item);
+  }
+};
```

Comment:
Should we add timestamp also to the requests and check that very old requests are not replayed but rather rejected.

It might also be useful to run `clearRetryQueue` at regular intervals to fail such old messages.

Reply:
The idea sounds good to me. I'm not yet familiar enough with the methods being called to specify an appropriate retirement time

Reply:
We can do retirement time, but running clearRetryQueue on an interval introduces high risk of rejecting something that was going to succeed otherwise.

clearRetryQueue is meant for recovery in catastrophic situations (like the UI being stuck and restarting)